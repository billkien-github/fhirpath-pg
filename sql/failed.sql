SELECT fhirpath_as_number('{"a":{"b": {"c": {"code":"foo"}}}}', '.a.b.c', 'Age', 'min') + 0.1;
SELECT fhirpath_as_number('{ "number":{"value": 42}, "date": {"value": "1980"}, "string": {"value": "Test"}, "token": {}, "reference": {}, "composite": {}, "quantity": {}, "uri": {}, "_empty_obj": {}, "_empty_arr": [], "_empty_str": "" }', '.number.value', 'integer', 'min');
SELECT fhirpath_as_number('{ "number":{"value": [1, 42, 5]}, "date": {"value": "1980"}, "string": {"value": "Test"}, "token": {}, "reference": {}, "composite": {}, "quantity": {}, "uri": {}, "_empty_obj": {}, "_empty_arr": [], "_empty_str": "" }', '.number.value', 'integer', 'min');
SELECT fhirpath_as_number('{"a":{"b": {"array": [40, 42, 39]}}}', '.a.b.array', 'integer', 'max') + 0.1;
