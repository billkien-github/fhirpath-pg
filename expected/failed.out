SELECT fhirpath_as_number('{"a":{"b": {"c": {"code":"foo"}}}}', '.a.b.c', 'Age', 'min') + 0.1;
 ?column? 
----------
         
(1 row)

SELECT fhirpath_as_number('{ "number":{"value": 42}, "date": {"value": "1980"}, "string": {"value": "Test"}, "token": {}, "reference": {}, "composite": {}, "quantity": {}, "uri": {}, "_empty_obj": {}, "_empty_arr": [], "_empty_str": "" }', '.number.value', 'integer', 'min');
 fhirpath_as_number 
--------------------
                 42
(1 row)

SELECT fhirpath_as_number('{ "number":{"value": [1, 42, 5]}, "date": {"value": "1980"}, "string": {"value": "Test"}, "token": {}, "reference": {}, "composite": {}, "quantity": {}, "uri": {}, "_empty_obj": {}, "_empty_arr": [], "_empty_str": "" }', '.number.value', 'integer', 'min');
 fhirpath_as_number 
--------------------
                  1
(1 row)

SELECT fhirpath_as_number('{"a":{"b": {"array": [40, 42, 39]}}}', '.a.b.array', 'decimal', 'max') + 0.1;
 fhirpath_as_number 
--------------------
                42.1
(1 row)
